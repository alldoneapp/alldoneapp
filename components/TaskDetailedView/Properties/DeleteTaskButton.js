import React from 'react'

import { colors } from '../../styles/global'
import store from '../../../redux/store'
import { showConfirmPopup } from '../../../redux/actions'
import { CONFIRM_POPUP_TRIGGER_DELETE_TASK } from '../../UIComponents/ConfirmPopup'
import Button from '../../UIControls/Button'
import { DV_TAB_ROOT_TASKS } from '../../../utils/TabNavigationConstants'
import { translate } from '../../../i18n/TranslationService'

export default function DeleteTaskButton({ task, projectId }) {
    const onPress = () => {
        store.dispatch(
            showConfirmPopup({
                trigger: CONFIRM_POPUP_TRIGGER_DELETE_TASK,
                object: {
                    task: task,
                    projectId: projectId,
                    navigation: DV_TAB_ROOT_TASKS,
                    originalTaskName: task.name,
                },
            })
        )
    }

    return (
        <Button
            icon={'trash-2'}
            title={translate(task.parentId ? 'Delete Subtask' : 'Delete Task')}
            type={'ghost'}
            iconColor={colors.UtilityRed200}
            titleStyle={{ color: colors.UtilityRed200 }}
            buttonStyle={{ borderColor: colors.UtilityRed200, borderWidth: 2 }}
            onPress={onPress}
            accessible={false}
        />
    )
}
