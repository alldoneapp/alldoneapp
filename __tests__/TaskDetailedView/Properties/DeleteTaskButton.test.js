import React from 'react'
import DeleteTaskButton from '../../../components/TaskDetailedView/Properties/DeleteTaskButton'

import renderer from 'react-test-renderer'

jest.mock('../../../utils/BackendBridge')
jest.mock('../../../utils/NavigationService')
jest.mock('firebase', () => ({ firestore: {} }));

describe('DeleteTaskButton component', () => {
    describe('DeleteTaskButton snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<DeleteTaskButton />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })

    describe('DeleteTaskButton methods', () => {
        it('should delete a task', () => {
            const tree = renderer.create(<DeleteTaskButton projectId="id0" task={{ id: 'id1' }} />)
            const instance = tree.getInstance()
            instance.onPress()
        })
    })
})
