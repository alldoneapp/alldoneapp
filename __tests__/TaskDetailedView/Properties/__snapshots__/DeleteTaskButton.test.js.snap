// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTaskButton component DeleteTaskButton snapshot test should render correctly 1`] = `
<View
  accessible={false}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#E00000",
      "borderRadius": 4,
      "borderWidth": 2,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 14,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#E00000",
            "fontFamily": "alldone",
            "fontSize": 20,
            "opacity": 1,
          }
        }
      >
        î¨–
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Medium",
          "fontSize": 14,
          "fontStyle": "normal",
          "letterSpacing": 0.8,
          "lineHeight": 14,
        },
        Object {
          "color": "#E00000",
        },
      ]
    }
  >
    Delete Task
  </Text>
</View>
`;
